packages/web/hooks/use-swap.ts

```diff
@@ -417,9 +421,41 @@ export function useSwap({
     quote,
   ]);
 
+  // Calculate token out fiat value from price impact and token in fiat value.
+  //
+  // This helps to mitigate the impact of various levels of caches. Here, we are guaranteed that to use
+  // the same fiat spot price used for both token in and token out amounts.
+  //
+  // The price impact is computed directly from quote, ensuring most up-to-date state.
+  // This guarantees consistency between token in and token out fiat values.
+  let tokenOutFiatValue: PricePretty = useMemo(
+    () =>
+      getTokenOutFiatValue(
+        quote?.priceImpactTokenOut?.toDec(),
+        inAmountInput.fiatValue?.toDec()
+      ),
+    [inAmountInput.fiatValue, quote?.priceImpactTokenOut]
+  );
```

Comment:
Note to reviewer: I think that we can refactor a lot of the webhooks in a similar manner to make them more testable. By extracting getters in such a way, we can define the invariants and cover them with unit tests to develop the foundation for making risky changes with more confidence.

Would appreciate feedback - maybe we could tackle some of these as part of stability?